### DISCUSSION TITLE
Docs code is not readable in production with vite-builder

### DISCUSSION THREAD

**Discussion by dohomi** [storybookjs/storybook#25025] [Help] [answered]
### Summary

Hey Guys,

I use Storybook v7 and vite for documenting a plugin I work on publicly. The documentation readme is not readable. 

Example: 
https://react-hook-form-material-ui.vercel.app/?path=/docs/autocomplete--docs
https://github.com/dohomi/react-hook-form-mui/blob/master/apps/storybook/stories/AutocompleteElement.stories.tsx

Do I miss anything in the Storybook config to create better output on production built? 
This is my config:
https://github.com/dohomi/react-hook-form-mui/blob/master/apps/storybook/.storybook/main.ts

<img width="1114" alt="Screenshot 2023-11-29 at 10 04 27" src="https://github.com/storybookjs/storybook/assets/489221/e631227a-1a88-43ca-8a4e-77f1f8ae7aeb">

### Create a reproduction

https://github.com/dohomi/react-hook-form-mui

**Comment by JReinhold** [comment, marked as answer]
I assume when you write "The documentation readme is not readable", you're referring to the component name being `[object Object]`?

I think this is because in your stories you're importing from your package using the workspace package resolution. This will make the builders treat it as an external package and import it as a `dist` file that gets minified and transpiled. Because of this, the docgen can't read the component name because it will get stripped out.
Alternatively it could also be the `forwardRef` that trips up the docgen pipeline.

There are two possible solutions here:

1. Add a `displayName` to all your components, these will not be stripped out when building.

Eg. [at this line](https://github.com/dohomi/react-hook-form-mui/blob/master/packages/rhf-mui/src/AutocompleteElement.tsx#L231) you would add `AutocompleteElement.displayName = "AutocompleteElement";`. The added benefit of this approach, is that any user using your library will see the correct component name when inspecting their applications in React DevTools, so this isn't just for Storybook.

4. You could merge your Storybook and the stories into your package, instead of having it as a separate package in your monorepo.

This way you can use relative imports in your stories, and the builder will not import the `dist`. This is similar to how you'd have unit tests next to your files, you wouldn't have unit tests in a separate package either. This also won't have an impact on your build output given that stories are still separate files not part of your bundle.

This won't solve the issue if the issue is coming from the `forwardRef` usage.

**Reply by dohomi** [reply]
Thanks for your detailed answer. I will try out the approach and definitely follow the suggestion to add a displayName for every component. Alternatively I will play around with the importing style. I will post my findings here

**Reply by dohomi** [reply]
OK I just made a fix with adding displayName to every component. Now the Stories are working well again thanks for your time
